### PR TITLE
Make minor workflow revisions and fix dedupe precommit

### PR DESCRIPTION
**Pull request by steverep**
## Proposed change

Steals the CI revisions from #14845 without the dedupe workflow plus a couple other fixes.  Summary:
- Removes setting CI environment variable (this is always set in actions)
- Add immutable flag to yarn install just to be safe
- Add concurrency rule to cancel in-progress CI runs when a new run is triggered
- Suppress ESLint warnings since only errors will fail
- Move duplicates check first to fail faster and prevent possibly only seeing ESLint or TS errors
- Give CI jobs friendlier names
- Fix an error in precommit hook not matching lock file

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Review by bramkragten — APPROVED** [review]

**Comment by steverep** [comment]
@bramkragten the CI is in a hung state because of the change in job names.  I think you might need to either force the merge or reset the protection rule.

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -15,8 +15,13 @@ env:
   NODE_OPTIONS: --max_old_space_size=6144
   GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
```
Comment (ludeeus) [inline .github/workflows/ci.yaml:19]: `github.event.pull_request.number ||` is redundant here and should be removed.

<https://docs.github.com/en/actions/learn-github-actions/contexts#github-context>
> The fully-formed ref of the branch or tag that triggered the workflow run. For workflows triggered by push, this is the branch or tag ref that was pushed. For workflows triggered by pull_request, this is the pull request merge branch. For workflows triggered by release, this is the release tag created. For other triggers, this is the branch or tag ref that triggered the workflow run. This is only set if a branch or tag is available for the event type. The ref given is fully-formed, meaning that for branches the format is refs/heads/<branch_name>, for pull requests it is refs/pull/<pr_number>/merge, and for tags it is refs/tags/<tag_name>. For example, refs/heads/feature-branch-1.

Comment (steverep) [reply .github/workflows/ci.yaml:19]: I just copied it from core for consistency

**Comment by steverep** [comment]
For whatever reason, it's still showing double CI jobs on new PRs now.  I think resetting the branch protection rule might fix it?

**Pull request merged**